gobblin-utility/src/main/java/gobblin/util/concurrent/TaskSchedulerFactory.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.util.concurrent;
+
+import org.apache.commons.lang.StringUtils;
+
+import com.google.common.base.Enums;
+import com.google.common.base.Optional;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+
+public class TaskSchedulerFactory {
+  @AllArgsConstructor
+  private enum TaskSchedulerType {
+    SCHEDULEDEXECUTORSERVICE(ScheduledExecutorServiceTaskScheduler.class),
+    HASHEDWHEELTIMER(HashedWheelTimerTaskScheduler.class);
+
+    @Getter
+    private final Class<? extends TaskScheduler> taskSchedulerClass;
+
+    public static Optional<TaskSchedulerType> parse(Optional<String> name) {
+      if (!name.isPresent() || StringUtils.isEmpty(name.get())) {
+        return Optional.absent();
+      }
+      return Enums.getIfPresent(TaskSchedulerType.class, name.get().toUpperCase());
+    }
+  }
+
+  private TaskSchedulerFactory() {
+  }
+
+  /**
+   * Gets an instance of the default {@link TaskScheduler} and ensures that it is started.
+   *
+   * @param <K> the type of the key for the {@link ScheduledTask}
+   * @param <T> the type of the {@link ScheduledTask}
+   * @return an instance of {@link TaskScheduler}
+   */
+  public static <K, T extends ScheduledTask<K>> TaskScheduler<K, T> get() {
+    return getImpl(Optional.<String>absent(), Optional.<String>absent());
+  }
+
+  /**
+   * Gets an instance of the default {@link TaskScheduler} and ensures that it is started.
+   *
+   * @param name the name associated threads created by the {@link TaskScheduler}
+   * @param <K> the type of the key for the {@link ScheduledTask}
+   * @param <T> the type of the {@link ScheduledTask}
+   * @return an instance of {@link TaskScheduler}
+   */
+  public static <K, T extends ScheduledTask<K>> TaskScheduler<K, T> get(Optional<String> name) {
+    return getImpl(Optional.<String>absent(), name);
+  }
+
+  /**
+   * Gets an instance of the {@link TaskScheduler} with the specified type and ensures that it is started. If
+   * the type is unknown an instance of the default {@link TaskScheduler} will be returned.
+   *
+   * @param type the type of the {@link TaskScheduler}
+   * @param <K> the type of the key for the {@link ScheduledTask}
+   * @param <T> the type of the {@link ScheduledTask}
+   * @return an instance of {@link TaskScheduler}
+   */
+  public static <K, T extends ScheduledTask<K>> TaskScheduler<K, T> get(String type) {
```

Comment:
This method's signature is too close to `get(Optional<String>)`, it will probably cause confusion in the future.